modules/sitemaps/sitemaps.php

```diff
@@ -202,6 +202,9 @@ function jetpack_load_xsl( $type = '' ) {
 
 	$transient_xsl = empty( $type ) ? 'jetpack_sitemap_xsl' : "jetpack_{$type}_sitemap_xsl";
 
+	// Add current language as suffix if WPML is active
+	$transient_xsl = $transient_xsl . wpml_transient_language();
```

Comment:
Please do—you can update this PR to do both add those filters and the functionality in the compat file.

If you could, please do add inline docs for the filters—that'd help the review process get knocked out faster! Thank you!